src/loader/bgw_launcher.c

```diff
@@ -332,15 +349,89 @@ db_hash_entry_create_if_not_exists(HTAB *db_htab, Oid db_oid)
 	return db_he;
 }
 
+/*
+ * Result from signalling a backend.
+ *
+ * Error codes are non-zero, and success is zero.
+ */
+enum SignalBackendResult
+{
+	SIGNAL_BACKEND_SUCCESS = 0,
+	SIGNAL_BACKEND_ERROR,
+	SIGNAL_BACKEND_NOPERMISSION,
+	SIGNAL_BACKEND_NOSUPERUSER,
+};
+
+/*
+ * Terminate a background worker.
+ *
+ * This is copied from pg_signal_backend() in
+ * src/backend/storage/ipc/signalfuncs.c but tweaked to not require a database
+ * connection since the launcher does not have one.
+ */
+static enum SignalBackendResult
+ts_signal_backend(int pid, int sig)
```

Comment:
Is it possible to get this code to run in a test?

Reply:
Not easily, no. It is just a copy of the `pg_signal_backend` code so it does not seems worth it to add a test for it.

Reply:
I am not so worried about this code being incorrect. But if it is never executed, there seems to be other paths calling this that are never executed either.

Reply:
I managed to extend the test to cover the situation described in the comment. Not running it as a flake test since it has a high risk of interfering with other tests and also setting the restart time to 10 seconds rather than the default 60 seconds.